skada/tests/test_mapping.py

```diff
@@ -157,22 +157,48 @@ def _base_test_new_X_adapt(estimator, da_dataset):
 
 
 @pytest.mark.parametrize(
-    "estimator",
+    "estimator, n_samples_source, n_samples_target",
     [
-        OTMappingAdapter(),
-        EntropicOTMappingAdapter(),
-        ClassRegularizerOTMappingAdapter(norm="lpl1"),
-        ClassRegularizerOTMappingAdapter(norm="l1l2"),
-        LinearOTMappingAdapter(),
-        CORALAdapter(),
-        pytest.param(
-            MMDLSConSMappingAdapter(gamma=1e-3),
-            marks=pytest.mark.skipif(not torch, reason="PyTorch not installed"),
+        (OTMappingAdapter(), 2, 1),
+        (OTMappingAdapter(), 1, 2),
+        (EntropicOTMappingAdapter(), 2, 1),
+        (EntropicOTMappingAdapter(), 1, 2),
+        (ClassRegularizerOTMappingAdapter(norm="lpl1"), 2, 1),
+        (ClassRegularizerOTMappingAdapter(norm="lpl1"), 1, 2),
+        (ClassRegularizerOTMappingAdapter(norm="l1l2"), 2, 1),
+        (ClassRegularizerOTMappingAdapter(norm="l1l2"), 1, 2),
+        (LinearOTMappingAdapter(), 2, 1),
+        (LinearOTMappingAdapter(), 1, 2),
+        (CORALAdapter(), 2, 1),
+        (CORALAdapter(), 1, 2),
+        (
+            pytest.param(
+                MMDLSConSMappingAdapter(gamma=1e-3),
+                1,
+                2,
+                marks=pytest.mark.skipif(not torch, reason="PyTorch not installed"),
+            )
+        ),
+        (
+            pytest.param(
+                MMDLSConSMappingAdapter(gamma=1e-3),
+                2,
+                1,
+                marks=pytest.mark.skipif(not torch, reason="PyTorch not installed"),
+            )
         ),
     ],
 )
-def test_new_X_adapt(estimator, da_dataset):
-    da_dataset = da_dataset.pack_train(as_sources=["s"], as_targets=["t"])
+def test_new_X_adapt(estimator, da_dataset, n_samples_source, n_samples_target):
+    da_dataset = make_shifted_datasets(
+        n_samples_source=n_samples_source,
+        n_samples_target=n_samples_target,
+        shift="concept_drift",
+        mean=0.5,
+        noise=0.3,
+        label="regression",
+        random_state=42,
+    )
```

Comment:
You throw away `da_dataset` and then generate a new dataset. IMO, you should modify the fixture in the `conftest.py` that outputs `da_dataset`, so that it outputs `da_datasets`, a list of datasets. Then, you do a for loop in the test. The overall idea is to generate data only in the `conftest.py`.